### PR TITLE
[Installer] Fixed permissions checking during installation

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | https://github.com/Sylius/Sylius/issues/3218
| License       | MIT
| Doc PR        |

There were some problem with permissions checking during running ``sylius:install:assets`` command. Now installer checks recursively permissions of ``web/bundles/`` and ``web/assets/`` directories and changes them if following directories are not writeable. If they cannot be changed, user gets message with specific directory that should have permissions changed manually.

### CODE REVIEW

File: src/Sylius/Bundle/InstallerBundle/Checker/CommandDirectoryCheckerInterface.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+namespace Sylius\Bundle\InstallerBundle\Checker;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+interface CommandDirectoryCheckerInterface
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Checker/CommandDirectoryCheckerInterface.php]: This interface is useless IMO.

File: src/Sylius/Bundle/InstallerBundle/Checker/CommandDirectoryChecker.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        $this->name = $name;
+    }
+
+    /**
+     * @param string          $directory
+     * @param OutputInterface $output
+     */
+    private function createDirectory($directory, OutputInterface $output)
+    {
+        if (!mkdir($directory, 0755, true)) {
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Checker/CommandDirectoryChecker.php]: Why not using Filesystem component from Symfony?

**Comment by michalmarcinkowski** [comment]
Thank you Mateusz

File: src/Sylius/Bundle/InstallerBundle/Checker/CommandDirectoryChecker.php [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+use Symfony\Component\Debug\Exception\ContextErrorException;
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class CommandDirectoryChecker
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/InstallerBundle/Checker/CommandDirectoryChecker.php]: CommandDirectoryChecker is a weird name. I had no idea what it does initially. It not only checks directories, but creates them and changes permissions. I think we should come up with a better naming for that class.

File: src/Sylius/Bundle/InstallerBundle/Resources/config/services.xml [hunk 4]
```diff
@@ -90,6 +91,10 @@
         <service id="sylius.installer.yaml_persister" class="%sylius.installer.yaml_persister.class%">
             <argument>%kernel.root_dir%/config/parameters.yml</argument>
         </service>
+
+        <service id="sylius.installer.checker.command_directory" class="%sylius.installer.checker.command_directory.class%">
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/InstallerBundle/Resources/config/services.xml]: Not sure about the service name. Will we have more checkers? (see comment above about checker name)

File: src/Sylius/Bundle/InstallerBundle/Checker/CommandDirectoryChecker.php (hunk 3 above)
Comment (Zales0123) [inline src/Sylius/Bundle/InstallerBundle/Checker/CommandDirectoryChecker.php]: You're bloody hell right! I just couldn't came out with more proper name, but I would definitely turn on my creativity in nearest future and push PR with some pretty good name for this service ;)

**Pull request merged**